### PR TITLE
Add policy set definition APIs

### PR DESCRIPTION
**Pull request by vivsriaus**
Add policy set definition APIs, add notScopes to policy assignment, add policyMode to policy definition, update the api-version

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Authorization/2017-06-01-preview/policy.json [hunk 1]
```diff
@@ -0,0 +1,960 @@
...
+          "200": {
+            "description": "OK - Returns information about the policy set definition.",
+            "schema": {
+              "$ref": "#/definitions/PolicySetDefinition"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policysetdefinitions": {
```
Comment (vivsriaus) [inline specification/resources/resource-manager/Microsoft.Authorization/2017-06-01-preview/policy.json]: @gandhiniraj @cyl3392207 please review the new APIs

**Comment by brendandburns** [comment]
LGTM

**Review by brendandburns — APPROVED** [review]

**Review by vishrutshah — CHANGES_REQUESTED** [review]
@vivsriaus This seems to be built upon 
`specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policy.json`. 

Could you please follow the contribution guidelines to expedite the review process. 
Guidelines:  https://github.com/Azure/azure-rest-api-specs/blob/current/.github/CONTRIBUTING.md#filenames-and-folder-structure
>>> It is time consuming to review the file line by line for every api-version. When you are creating the swagger spec for the new api-version, please copy the swagger spec from the previous version in to the new api-versioned folder and commit it. After that overwrite it with the changes for the new api-version. This makes it easy for us to review the changes.

- [ ] Has this been reviewed by ARM team? 

Thanks

**Comment by vivsriaus** [comment]
@vishrutshah Makes sense. I'll close this PR. New one is [here](https://github.com/Azure/azure-rest-api-specs/pull/1587)

**Pull request closed without merging**